public/js/extraFunctions.js

```diff
@@ -654,6 +654,143 @@ function tableData_closeAllSelect(elmnt) {
 
 }
 
+let initialize_resize_hamburger = function() {
+
+  //total width of all items in tab minus hamburger
+  let total_width = -44.25;
+
+  //width of all items to the left (because they get removed last) plus logout button and hamburger widths
+  //44.25 is hamburger width, has to be hardcoded becuase it might be display: none;'d
+  let left_width = 44.25 + $('#logout_button').outerWidth()
+
+  //gets all non-tablinks-right elements and adds their width to total_width  and also left_width
+  $('.tab .tablinks:not(.tablinks-right)').outerWidth(function(_i, w) {total_width += w; left_width += w;});
+
+  //gets all tablinks-right elements and adds their width to total_width
+  $('.tab .tablinks-right').outerWidth(function(_i, w) {total_width += w;})
+
+  const switch_left_items = function() {
+    //checks if left items are in sidenav
+    const in_sidenav = $('.tab .tablinks:not(.tablinks-right, .switch-exempt)').length == 0
+
+    if (in_sidenav) {
+      //moves items from sidenav to tab
+      $('#sidenav .tablinks:not(.tablinks-right, .switch-exempt)').detach().appendTo($(".tab"));
+    } else {
+      //moves items from tab to sidenav
+      $('.tab .tablinks:not(.tablinks-right, .switch-exempt)').detach().appendTo($("#sidenav"));
+      //puts all the tablinks-right things below the non-tablinks-right things
+      for (const child of $('#sidenav .tablinks-right')) {
+        $('#sidenav').append(child);
+      }
+    }
+  }
+
+  //REASON IT ADDS ITEMS IN REVERSE:
+  //otherwise the items would basically go from right to left (in tab) 
+  //to down to up, when I'd like it to be up to down.
+  const switch_right_items = function() {
+    //checks if right-items (more specifically buttons) are hidden
+
+    const in_sidenav = $('.tab .gpa_custom-select').length == 0
+
+    if (in_sidenav) {
+
+      //adds right items to tab in reverse
+      const children = $('#sidenav .tablinks-right:not(#logout_button)')
+      for (i=0; i<children.length; i++) {
+        $(".tab").append(children[children.length-1-i])
+      }
```

Comment:
Adding the elements in reverse can be done in a more intuitive, more idiomatic, and less error-prone way than a C-style `for` loop:
```suggestion
      for (const child of [...children].reverse()) {
        $(".tab").append(child);
      }
```
Here, `[...children]` converts `children` to an array. One can then iterate over it in reverse order using a `for`…`of` loop and `Array.reverse()`.